ui/src/app/pages/workspace/workspace-share.tsx

```diff
@@ -322,20 +351,19 @@ export const WorkspaceShare = fp.flow(withUserProfile())(
         });
     }
 
-    setRole(e, user): void {
-      const oldUserRoles = this.state.userRoles;
-      const newUserRoleList = fp.map((u) => {
+    setRole(e, user: UserRole): void {
+      const userRoles = fp.map((u) => {
```

Comment:
userRoles is what is displayed in the modal: simply update the existing collab.

userRolesToChange is the changeset we use in the share request. Most likely, it's not there yet. But we may have already added this user to userRolesToChange, so we need to filter this out first.